ecs-update/ecs-update.go

```diff
@@ -856,6 +926,30 @@ func (e *packageEditor) modifyManifest() error {
 		e.result.Manifest.FormatVersionNew = e.config.Manifest.FormatVersion
 	}
 
+	if e.config.Manifest.KibanaVersion != "" && e.config.Manifest.KibanaVersion != e.pkg.Manifest.Conditions.Kibana.Version {
+		var haveStack, wantStack versionConstraint
+		err = haveStack.Set(e.pkg.Manifest.Conditions.Kibana.Version)
```

Comment:
Values like this https://github.com/elastic/integrations/blob/0cd817a50cfd03e49b1dbd4d21b8daa91b0e48a3/packages/fireeye/manifest.yml#L12 fail to parse.

```json
  {
    "package": "fireeye",
    "changed": false,
    "failed": true,
    "error": "failed to parse existing version: strconv.ParseInt: parsing \"0 || ^8.0.0\": invalid syntax"
  }
```

The package-registry uses https://pkg.go.dev/github.com/Masterminds/semver/v3#Constraints to parse this. We should either be able to accept the same format or use that library. If you use that library I'm not sure you can implement an equivalent `haveStack.LessThan(want)` check to determine if it needs raised.

<details>

<summary>Breadcrumbs to find the version constraint format used in packages</summary>

- https://github.com/elastic/package-spec/blob/a895b110b5a86b6ae5410c8bd3834539e27a1b92/spec/integration/manifest.spec.yml#L140
- https://github.com/elastic/package-spec/blob/a895b110b5a86b6ae5410c8bd3834539e27a1b92/code/go/internal/validator/semantic/validate_minimum_kibana_version.go#L151-L164
- https://github.com/elastic/package-registry/blob/1c6fed752fa5c55ffed40b50805bffac3fddd21f/packages/package.go#L121
- https://pkg.go.dev/github.com/Masterminds/semver/v3#Constraints

</details>

Reply:
Not yet correctly handled.